src/sphinx_book_theme/_transforms.py

```diff
@@ -0,0 +1,70 @@
+from sphinx.transforms.post_transforms import SphinxPostTransform
+from typing import Any
+from docutils import nodes as docutil_nodes
+from sphinx import addnodes as sphinx_nodes
+from .nodes import SideNoteNode
+import copy
+
+
+class HandleFootnoteTransform(SphinxPostTransform):
+
+    default_priority = 1
+    formats = ("html",)
+
+    def run(self, **kwargs: Any) -> None:
+        theme_options = self.env.config.html_theme_options
+        if theme_options.get("use_sidenotes", False) is False:
+            return None
+        for node in self.document.traverse(docutil_nodes.footnote_reference):
+            parent = None
+            for ftnode in self.document.traverse(docutil_nodes.footnote):
+                # matching the footnote reference with footnote
+                if (
+                    len(ftnode.attributes["backrefs"])
+                    and ftnode.attributes["backrefs"][0] == node.attributes["ids"][0]
+                ):
+                    parent = ftnode.parent
+                    text = ftnode.children[1].astext()
+
+                    sidenote = SideNoteNode()
+                    para = docutil_nodes.inline()
+                    label = ftnode.children[0].astext()
+
+                    if text.startswith("{-}"):
+                        # marginnotes will have content starting with {-}
+                        # remove the number so it doesn't show
+                        para.attributes["classes"].append("marginnote")
+                        para.append(docutil_nodes.Text(text.replace("{-}", "")))
```

Comment:
Should we do a `.strip()` here as well since there will be a space after the `{-}`?

Reply:
We can I guess. But without `strip()` it seems to be stripping spaces as well, no matter how much space we give.

Reply:
fair enough - maybe this is a sphinx parsing thing under the hood